src/main/kotlin/oop/Memento/CareTaker.kt

```diff
@@ -1,5 +1,11 @@
 package oop.Memento
 
 class CareTaker<T> {
-  val mementoList = mutableListOf<Memento<T>>()
+  private val mementoList = mutableListOf<Memento<T>>()
+
+  fun addMemento(memento: Memento<T>) {
+    mementoList.add(memento)
+  }
+
+  fun getMemento(index: Int) : Memento<T>? = mementoList[index]
 }
```

Comment:
Is this really the best way to do this in Kotlin? I see little sense in wrapping a mutable list 🤔